### ISSUE TITLE
Windows 10 installation issues

### ISSUE THREAD

**Issue by Brorean** [xournalpp/xournalpp#1600]
Just tried the Windows installer on Windows 10 (1903) and it appears to be not working correctly.
First the installer does not add the program to the start menu.
After installation when trying to run the program manually from `C:\Program Files\Xournal++\bin\xournalpp.exe` you get two System Errors stating that `The code execution cannot proceed because libcrypto-1_1-x64.dll was not found.  Reinstalling the program may fix this problem.` and another one for libssl-1_1-x64.dll.

**Comment by qukequke** [comment]
![image](https://user-images.githubusercontent.com/33408558/68655098-5d682500-056a-11ea-8546-a3418ec71d38.png)
same！

**Comment by LittleHuba** [comment]
Seems like we miss a few libraries in the packaging. I'll look into it.

**Comment by Technius** [comment]
Might be a CI configuration issue. I built a version locally in MSYS and it copied the two mentioned libraries correctly. The output of `ldd build/src/xournalpp.exe` should contain `/mingw64/bin/librypto-1_1-x64.dll` (as well as libssl).

**Comment by LittleHuba** [comment]
It could be that the files are missing in the MSYS image used for the CI pipeline. But I'm not sure why if that is the case.
As you have a working Windows machine @Technius could you check this out? It is quite a hassle for me as I only have my small Linux laptop available currently.

**Comment by agusmba** [comment]
![image](https://user-images.githubusercontent.com/3104020/68807336-f03ab980-0667-11ea-9ca4-ed329e6a0d8e.png)

In my case the missing library seems to be a different one (actually it looks like a version mismatch instead of a missing lib)

**Comment by LittleHuba** [comment]
It could be as the CI pipeline is building on a Server OS that different libraries are packaged (just a guess).But that would explain why everything works if it is packaged on a normal Win10 installation. We might have to shift the Release pipeline to a Desktop OS version to fix it then.

**Comment by LittleHuba** [comment]
> The output of ldd build/src/xournalpp.exe should contain /mingw64/bin/librypto-1_1-x64.dll (as well as libssl).

The output is actually in the `ldd` call but is filtered out since the Windows version we use for the CI is indeed a Server version which includes these libraries.

@agusmba Can you check if the file exists at the directory the dialog lists?

**Comment by LittleHuba** [comment]
#1628 shoud fix the problem for @Brorean and @qukequke.

Please check out the setup binary [here](https://dev.azure.com/xournalpp/xournalpp/xournalppTeam/_build/results?buildId=3559&view=artifacts), (the correct artifact is under "windows") and report if it works for you.

@agusmba please also check this binary as it seems like your error is generated when the openssl libaries used for building are incompatible with the ones used for running. Including these libraries might fix the issue.

**Comment by agusmba** [comment]
@LittleHuba I tried the new setup, and now I get several more errors:

missing libgdk-3-0.dll, libgtk-3-0.dll, libportaudio-2.dll, libpoppler-glib-8.dll

**Comment by qukequke** [comment]
@LittleHuba  I tried too. And I got the same question as @agusmba .

**Comment by LittleHuba** [comment]
Perfect so we do have the right way to solve this. 
The libraries you get now are the locally generated and internally in msys installed ones. Most likely it is also some kind of path issue.

**Comment by LittleHuba** [comment]
Please try again with [this](https://dev.azure.com/xournalpp/xournalpp/_build/results?buildId=3576&view=artifacts) version

**Comment by agusmba** [comment]
I'll test it tonight at home, however looking at the change you made in the last version, I think you might be missing some options for grep:

```diff
- ldd ../build/src/xournalpp.exe | grep '\/mingw.*\.dll|\/c/windows/SYSTEM32/lib.*\.dll' -o
+ ldd ../build/src/xournalpp.exe | grep -iE '\/mingw.*\.dll|\/c/windows/SYSTEM32/lib.*\.dll' -o
```

I don't think you need to backslash your forward slashes, but anyway, `-i` ignores case, because sometimes the library paths are `Windows` or `windows`..., and `-E` allows you to use the OR `|` in the regexp

**Comment by LittleHuba** [comment]
The grep on mingw works quite awkwardly. And since testing one option takes me about half an hour I just sticked to the safe and easy way.
You can provide the short version as a tested PR if you want. You have a Windows system and can test it on that. Doing this on the pipelines is a painful job.

**Comment by Brorean** [comment]
> 
> 
> Please try again with [this](https://dev.azure.com/xournalpp/xournalpp/_build/results?buildId=3576&view=artifacts) version

This version started just fine on my version of Windows and didn't display any error messages.  However I was unable to write anything (using mouse or pen), type anything, or insert an image; even if I ran the application as administrator.

**Comment by LittleHuba** [comment]
Could you start the application from a console and send the output?

**Comment by Brorean** [comment]
I opened up a command prompt and ran `xournalapp.exe`; the application started but no log messages were written to the screen.  Also immediately after running the application I got the command line prompt to run another command; I attached a screenshot to show you since I can't really explain that well.
Did I do what you wanted?  Or is there some command line flag I need to use?

![Capture](https://user-images.githubusercontent.com/6954891/69655847-0cadfd00-103d-11ea-91e9-67f4d4966d6c.PNG)

**Comment by LittleHuba** [comment]
You will have to install MSYS then and build it localy on your machine. Otherwise you will not get any debug messages as it seems. Check out our Wiki to see how all that is done.

Did prior versions work?

**Comment by agusmba** [comment]
I can confirm Brorean's findings, the application starts, but I can't edit anything on the page.

Console output is not very helpful:

```
$ ./xournalpp_bin.exe

** (xournalpp_bin.exe:18548): WARNING **: 19:55:00.274: Could not open plugin dir: «C:\Program Files\Xournal++\bin\..\plugins»

(xournalpp_bin.exe:18548): Gtk-WARNING **: 19:55:00.621: Theme parsing error: <data>:1:22: Junk at end of value for font-family

(xournalpp_bin.exe:18548): Pango-WARNING **: 19:55:00.721: couldn't load font "C059 Not-Rotated 8", falling back to "Sans Not-Rotated 8", expect ugly output.
```

Trying to use the different tools (pen, text...) yield no errors, but nothing is edited on the page.

The previous release available on github worked fine.

**Comment by LittleHuba** [comment]
Could you check in the settings if the new input system is activated? If yes check out the device classes and set them to a good value. I think this might be the issue.

**Comment by agusmba** [comment]
You were right, the new input system was active, but the selected device class was "None" 🤦‍♂ 
After selecting "Mouse" it works.

**Comment by LittleHuba** [comment]
Could you tell me what class the mouse was detected as? In the brackets after the name. We might have to fix that before our next release or we will get some hundreds of complaints...

**Comment by agusmba** [comment]
![image](https://user-images.githubusercontent.com/3104020/69759227-dc13b380-1161-11ea-969f-3f7a3000eb0e.png)

Maybe a case-mismatch issue?
I tried uninstalling and re-installing but I don't get the original problem (maybe I should delete my preferences wherever they might be?)

**Comment by LittleHuba** [comment]
This should have been automatically matched. Could you create a new issue for that?
Yeah the matching is stored in the settings file. Not quite sure where this is stored on Windows. I assume somewhere in your user directory. Maybe AppData?

**Comment by agusmba** [comment]
I'll try looking for it tonight to see if I can delete it and replicate the matching problem.

**Comment by agusmba** [comment]
Mmmm, I couldn't test it on my Win10 Home edition yesterday (I haven't forgotten, I'll open a new issue if there is a mismatch on a default installation), however I just tried it on a Windows 10 Enterprise and got another missing dll error: `VCRUNTIME140.dll`

**Comment by LittleHuba** [comment]
You'll have to do it on the same machine. Edit the settings file and remove the section with your device name in it.

The runtime should be installed on the machine itself. We need to add this to the installer. Shipping it is not an option if I remember correctly.

**Comment by LittleHuba** [comment]
To install the Visual Studio C++ runtime 14.0 we will have to ship the redistributable installer. Installing it silently in the background using NSIS is not a problem. We only have to search for the correct version of the redistributable installer and add the logic into the pipeline to add it.

**Comment by LittleHuba** [comment]
@agusmba could you check if this installer fixes the missing library?
https://aka.ms/vs/16/release/vc_redist.x86.exe

**Comment by agusmba** [comment]
> could you check if this installer fixes the missing library?
> https://aka.ms/vs/16/release/vc_redist.x86.exe

Hi, I tested this on the Win10Enterprise box, but I still got the missing dll error. Note that this is not my regular home/work computer, just a virtualbox I fired up to test xournalpp installer.

However, you pointed me towards trying to install the [VC redist libraries for 64bit systems](https://aka.ms/vs/16/release/vc_redist.x64.exe).
After that, xournalpp opened fine (with the #1639 issue as expected) :clap:

**Comment by LittleHuba** [comment]
Perfect, yeah did not think about architecture matching. 64-bit makes more sense...

Okay I'll add the installer to our own installer later on. Thanks for testing it!

**Comment by andreasb242** [comment]
Hi together

I did some checks, and I'm really sure, that LittleHuba misinterpreted some stuff here.
I checked and I'm currently really sure, that Xournal++ itself does not require any Visual Studio DLLs, as we are not using Visual Studio.

The Problem is caused by the Windows DLL Loading, good known as DLL-Hell.
I currently cannot reproduce the issue on my Windows devices, therefore I need some help from somebody with the Problem.

Can please somebody download the Dependency Walker from:
http://www.dependencywalker.com/ (Yes, looks really crusty, also sometimes not easy to use, but the best tool I know for this)
And open xournalpp_bin.exe, when it's loaded, save the output as .txt and upload it here?

Then we can see, why the DLL is loaded, or missing.
If you need help, you can ask here or write me directly in our chat:
https://gitter.im/xournalpp/xournalpp

I expect, that there is another GTK Application installed, which was compiled with Visual C++. But lets see ;-)

**Comment by agusmba** [comment]
Do you want the regular txt or the one saved with lists?

**Comment by agusmba** [comment]
This is the regular txt exported:

[xournalpp_bin.zip](https://github.com/xournalpp/xournalpp/files/3954536/xournalpp_bin.zip)

**Comment by andreasb242** [comment]
Thanks!

But one issue: VCRUNTIME140.dll is not listed, so it's not used, at least not statically.
This makes it more complicated.
When did the error occur, directly on the start?

@agusmba You got this error on Windows 10 Enterprise, was this a fresh install, or a system where already a lot of software was installed?

**Comment by agusmba** [comment]
The error occurred right at the start (no xournalpp windows was ever shown).

> @agusmba You got this error on Windows 10 Enterprise, was this a fresh install, or a system where already a lot of software was installed?

It was an oldish (>1 y.o.) Microsoft Edge testing [VM](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/), so the image could be missing *standard* libraries present in normal OS installs. There isn't much installed on it (git, different git GUIs, vscode, pandoc, mingw).

**Comment by xx299x** [comment]
The problem still exists after I install the vc_redist.x86.exe
![image_283](https://user-images.githubusercontent.com/23482348/71309803-570d7880-2447-11ea-82ad-87b18f83d4ff.png)
![image_285](https://user-images.githubusercontent.com/23482348/71309805-58d73c00-2447-11ea-85af-54fc8116a9d1.png)


> @agusmba could you check if this installer fixes the missing library?
> https://aka.ms/vs/16/release/vc_redist.x86.exe
> 
> This is version 2019 if it does not ship v14 we will need version 2013 of the installer. The webpage was not very conclusive on if it does or not.

**Comment by Febbe** [comment]
@xx299x as far as I know, @LittleHuba fixed that in the newest release. So please try it that way. Nevertheless, installing ms vc_redist.x64, shouldn't be the solution, since the missing libraries here are part of OpenSSL. (The version provided by Vc_redist could link against the dynamic runtime environment of msvc). Just installing OpenSSL and adding the binary path to the PATH environment variable should help.

**Comment by xx299x** [comment]
> @xx299x as far as I know, @LittleHuba fixed that in the newest release. So please try it that way. Nevertheless, installing ms vc_redist.x64, shouldn't be the solution, since the missing libraries here are part of OpenSSL. (The version provided by Vc_redist could link against the dynamic runtime environment of msvc). Just installing OpenSSL and adding the binary path to the PATH environment variable should help.

thank you！It is woring after I install openssl.

**Comment by andreasb242** [comment]
Ca you also provide the output of Dependency Walker? See my comment above. Thanks!

**Comment by xx299x** [comment]
> Ca you also provide the output of Dependency Walker? See my comment above. Thanks!

Of course,but I happened some problems as follow.
![image_287](https://user-images.githubusercontent.com/23482348/71408260-822ade80-2678-11ea-97ed-2f166c533b1f.png)

**Comment by andreasb242** [comment]
Before you open the exe, there is a Path config, where depends searchs for .dlls, I usually throw out the applications folder there, which are not required, and then try again.
(But you risk to loose an important entry... But I have no better option...)

**Comment by xx299x** [comment]
> Before you open the exe, there is a Path config, where depends searchs for .dlls, I usually throw out the applications folder there, which are not required, and then try again.
> (But you risk to loose an important entry... But I have no better option...)

emmm....This the fileis of txt too large!
![image](https://user-images.githubusercontent.com/23482348/71410291-a211d080-267f-11ea-8d84-fc591bbf72a2.png)

**Comment by pkeir** [comment]
I install Xournal++ (1.0.16) on Windows 10 today for the first time, but a message box reports that libcrypto-1_1-x64.dll and libssl-1_1-x64.dll are missing.

**Comment by jellespijker** [comment]
Installing the windows OpenSSL, which can be found at https://slproweb.com/products/Win32OpenSSL.html , solved the missing crypto and libssl warnings for me

**Comment by Technius** [comment]
I'm guessing the issue has been that the CI image already has libcrypto and libssl, so the packaging scripts have not been copying them. I just pushed out a fix attempt, and we'll see if it works after the nightly is built tomorrow.

**Comment by nicegamer7** [comment]
Your fix works. I tested the latest nightly on two systems and it worked. Thanks.

**Comment by zingmars** [comment]
Can confirm, latest version works ootb for me too. Didn't work before.

**Comment by Technius** [comment]
Fixed in 1.0.17.
